src/core/ucc_team.h

```diff
@@ -41,4 +51,44 @@ typedef struct ucc_team {
 #define UCC_TEAM_ID_MAX ((uint16_t)UCC_BIT(15) - 1)
 
 void ucc_copy_team_params(ucc_team_params_t *dst, const ucc_team_params_t *src);
+
+/* Returns addressing information for "rank" in a team.
+   If ucc context was created with OOB then addr storage is located on context.
+   In that case we need to map rank to ctx_rank first. Otherwise, addr
+   storage is per-team: just use rank then.
+
+   The returned value is "header": it stores proc_info, ctx_id and addresses
+   of TL/CL components.*/
+static inline ucc_context_addr_header_t *
+ucc_get_team_ep_header(ucc_context_t *context, ucc_team_t *team,
+                       ucc_rank_t rank)
+{
+    ucc_addr_storage_t *storage      = context->addr_storage.storage
+                                           ? &context->addr_storage
+                                           : &team->addr_storage;
+    ucc_rank_t          storage_rank = context->addr_storage.storage
+                                           ? ucc_ep_map_eval(team->ctx_map, rank)
+                                           : rank;
+
+    return UCC_ADDR_STORAGE_RANK_HEADER(storage, storage_rank);
+}
+
+/* Gets the component specific address of rank in a team.
+   First we get the header, and then find the component address
+   by offset */
+static inline void *ucc_get_team_ep_addr(ucc_context_t *context,
+                                         ucc_team_t *team, ucc_rank_t rank,
+                                         unsigned long component_id)
+{
+    ucc_context_addr_header_t *h = ucc_get_team_ep_header(context, team, rank);
+    int                        i;
+    for (i = 0; i < h->n_components; i++) {
+        if (h->components[i].id == component_id) {
+            return PTR_OFFSET(h, h->components[i].offset);
+        }
+    }
+    ucc_assert(0);
```

Comment:
Remove

Reply:
wanted to make it readable that we never get there - i'll change the assertion on positive